ambry-commons/src/main/java/com.github.ambry.commons/ResponseHandler.java

```diff
@@ -53,9 +53,18 @@ private void onServerEvent(ReplicaId replicaId, ServerErrorCode errorCode) {
         break;
       case Partition_ReadOnly:
         clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Partition_ReadOnly);
-        //fall through
+        clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Disk_Ok);
+        clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Replica_Available);
+        break;
+      case Temporarily_Disabled:
+      case Replica_Unavailable:
+        clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Replica_Unavailable);
+        clusterMap.onReplicaEvent(replicaId, ReplicaEventType.Disk_Ok);
```

Comment:
Not entirely sure we should generate `Disk_Ok` for this. Will discuss with you.

Reply:
Hmm, I agree with you, maybe we should prioritize the error code.

Reply:
Is disk availability the first thing that gets checked? Perhaps it might be okay to just fire replica unavailability event here and not make any other assumptions.

Reply:
Right, there might be a case where all replicas on the disk are unavailable due to disk failure. In this case, `Disk_Ok` is not supposed to be the event.